src/index.js

```diff
@@ -0,0 +1,288 @@
+  addEscapeListener();
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  removeEscapeListener();
+}
+
+// Form handlers
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  editSubmitButton.textContent = "Saving...";
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+      confirmTitle.textContent = "Edit Avatar";
+      confirmMessage.textContent = "You have successfully Updated a Avatar";
+      openModal(confirmModal);
+      setTimeout(function () {
+        closeModal(confirmModal);
+      }, 2000);
+    })
+    .catch(console.error)
+    .finally(() => {
+      editSubmitButton.textContent = "Save";
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  cardSubmitButton.textContent = "Saving...";
+  api
+    .addNewCard(inputValues)
+    .then((card) => {
+      const cardElement = getCardElement(card);
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      cardForm.reset();
+      disableButton(evt.submitter, settings);
+      confirmTitle.textContent = "New Post";
+      confirmMessage.textContent = "You have successfully created a post";
+      openModal(confirmModal);
+      setTimeout(function () {
+        closeModal(confirmModal);
+      }, 2000);
+    })
+    .catch(console.error)
+    .finally(() => {
+      cardSubmitButton.textContent = "Save";
+    });
+}
+
+changeAvatarForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+
+  const formData = new FormData(event.target);
+  avatarSubmitButton.textContent = "Saving...";
+  const data = Object.fromEntries(formData);
+  api
+    .editAvatar(data)
+    .then(() => {
+      const avatar = document.querySelector(".profile__avatar");
+      avatar.src = data.avatar;
+      closeModal(changeAvatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      avatarSubmitButton.textContent = "Save"; // Reset to default text
+    });
+});
+
+// Card element generator
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImgEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  cardImgEl.src = data.link;
+  cardImgEl.alt = data.name;
+  cardNameEl.textContent = data.name;
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardLikeBtn.addEventListener("click", () => {
+    const toggleLike = cardLikeBtn.classList.toggle("card__like-btn_liked");
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.